compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -5604,6 +5624,61 @@ private BType resolveQueryType(SymbolEnv env, BLangExpression selectExp, BType c
         }
     }
 
+    private BType getMapType(BType selectType, Location pos) {
+        BType referredType = Types.getReferredType(selectType);
+        if (referredType.tag == TypeTags.INTERSECTION) {
+            referredType = ((BIntersectionType) referredType).effectiveType;
+        }
+
+        if (referredType.tag == TypeTags.UNION) {
+            BUnionType unionType = (BUnionType) referredType;
+            LinkedHashSet<BType> memberTypes = new LinkedHashSet<>();
+            for (BType type : unionType.getMemberTypes()) {
+                BType mapType = getTypeForMap(type, pos);
+                if (mapType == null) {
+                    return symTable.noType;
+                }
+                memberTypes.add(mapType);
+            }
+            return new BUnionType(null, memberTypes, false, false);
+        } else {
+            BType mapType = getTypeForMap(referredType, pos);
```

Comment:
We can return `symTable.noType` or `symTable.semanticError` for null cases and remove this null check.